pep-0xxx.rst

```diff
@@ -0,0 +1,1250 @@
+
+            return super().__new__(
+                cls, validate_certificates, certificate_chain, ciphers,
+                inner_protocols, lowest_supported_version,
+                highest_supported_version, trust_store, sni_callback
+            )
+
+        def update(self, validate_certificates=_DEFAULT_VALUE,
+                         certificate_chain=_DEFAULT_VALUE,
+                         ciphers=_DEFAULT_VALUE,
+                         inner_protocols=_DEFAULT_VALUE,
+                         lowest_supported_version=_DEFAULT_VALUE,
+                         highest_supported_version=_DEFAULT_VALUE,
+                         trust_store=_DEFAULT_VALUE,
+                         sni_callback=_DEFAULT_VALUE):
+            """
+            Create a new ``TLSConfiguration``, overriding some of the settings
+            on the original configuration with the new settings.
+            """
+            if validate_certificates is _DEFAULT_VALUE:
+                validate_certificates = self.validate_certificates
+
+            if certificate_chain is _DEFAULT_VALUE:
+                certificate_chain = self.certificate_chain
+
+            if ciphers is _DEFAULT_VALUE:
+                ciphers = self.ciphers
+
+            if inner_protocols is _DEFAULT_VALUE:
+                inner_protocols = self.inner_protocols
+
+            if lowest_supported_version is _DEFAULT_VALUE:
+                lowest_supported_version = self.lowest_supported_version
+
+            if highest_supported_version is _DEFAULT_VALUE:
+                highest_supported_version = self.highest_supported_version
+
+            if trust_store is _DEFAULT_VALUE:
+                trust_store = self.trust_store
+
+            if sni_callback is _DEFAULT_VALUE:
+                sni_callback = self.sni_callback
+
+            return self.__class__(
+                validate_certificates, certificate_chain, ciphers,
+                inner_protocols, lowest_supported_version,
+                highest_supported_version, trust_store, sni_callback
+            )
+
+
+
+Context
+~~~~~~~
+
+We define two Context abstract base classes. These ABCs define objects that
+allow configuration of TLS to be applied to specific connections. They can be
+thought of as factories for ``TLSWrappedSocket`` and ``TLSWrappedBuffer``
+objects.
+
+Unlike the current ``ssl`` module, we provide two context classes instead of
+one. Specifically, we provide the ``ClientContext`` and ``ServerContext``
+classes. This simplifies the APIs (for example, there is no sense in the server
+providing the ``server_hostname`` parameter to ``ssl.SSLContext.wrap_socket``,
+but because there is only one context class that parameter is still available),
+and ensures that implementations know as early as possible which side of a TLS
+connection they will serve. Additionally, it allows implementations to opt-out
+of one or either side of the connection. For example, SChannel on macOS is not
```

Comment:
I’m sorry I didn’t notice this before re-reading on python-ideas but I’m sure you mean SecureTransport here (and later in this paragraph).  SChannel is a Windows API.